### PR TITLE
checker: don't disallow defining methods on interfaces

### PR DESCRIPTION
**Pull request by spaceface777**
As discussed on Discord- this error doesn't really make much sense, and this can actually be very useful, especially now that we have interface fields.

### CODE REVIEW

File: vlib/v/checker/tests/no_method_on_interface_propagation.vv [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+fn (a Animal) foo() {}
+
+fn new_animal(breed string) Animal {
+	return &Cat{ breed }
+}
+
+fn test_methods_on_interfaces_dont_exist_on_implementers() {
+	mut a := new_animal('persian')
+    if a is Cat {
```
Comment (medvednikov) [inline vlib/v/checker/tests/no_method_on_interface_propagation.vv:17]: vfmt

**Comment by medvednikov** [comment]
@spaceface777 can you also document this please?

**Comment by danieldaeschle** [comment]
Can anyone explain to me how it makes sense defining methods on interfaces? Which languages supports this?? An interface is a shape for a struct and no value to be used...

**Comment by spaceface777** [comment]
this has been discussed several times already on Discord; here's one discussion from yesterday:
https://discord.com/channels/592103645835821068/592320321995014154/803641864595046440

**Pull request merged**